### PR TITLE
Feature/eui 924 media viewer aa

### PR DESCRIPTION
**Pull request by adnanakgun**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-924

### Change description ###

Too many to count

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: api/amendedJurisdictions/index.ts [hunk 1]
```diff
@@ -4,26 +4,33 @@ import { config } from '../config'
 import { http } from '../lib/http'
 import { setHeaders } from '../lib/proxy'
 
+/**
+ * Manually filtering returned jurisdictions
+ * to make available jurisdiction in filters array only
+ */
 export async function getJurisdictions(req: express.Request, res: express.Response) {
+    console.log(req.url)
```
Comment (ronaldmansveld) [inline api/amendedJurisdictions/index.ts]: Do we want to keep stray console.logs in the code?

File: api/amendedJurisdictions/index.ts [hunk 2]
```diff
@@ -4,26 +4,33 @@ import { config } from '../config'
...
     let url = striptags(req.url)
+    console.log(url)
```
Comment (ronaldmansveld) [inline api/amendedJurisdictions/index.ts]: As above

File: api/emAnno/emAnnoService.spec.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
+/* tslint:disable:no-unused-expression no-var-requires */
```
Comment (ronaldmansveld) [inline api/emAnno/emAnnoService.spec.ts]: Is there a reason to disable the no-unused-expression rule?

File: api/emAnno/emAnnoService.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+const logger: JUILogger = log4jui.getLogger('em-anno-service')
+
+/**
+ * Get Annotations
+ *
+ * @param annotationsPath
+ * @returns {Promise<null>}
+ */
+export async function handleGet(annotationsPath) {
```
Comment (ronaldmansveld) [inline api/emAnno/emAnnoService.ts]: Missing typing

File: api/emAnno/emAnnoService.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    try {
+        logger.info('getting annotations', annotationsPath)
+        const response = await http.get(annotationsPath)
```
Comment (ronaldmansveld) [inline api/emAnno/emAnnoService.ts]: Missing typing (what model does that http call return, what is the format of the data property that gets returned?)

File: api/emAnno/emAnnoService.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * @param annotationsPath
+ * @param body
+ * @returns {Promise<null>}
+ */
+export async function handlePost(annotationsPath, body) {
+
+    try {
+        logger.info('posting annotations', annotationsPath)
+        const response = await http.post(annotationsPath, body)
```
Comment (ronaldmansveld) [inline api/emAnno/emAnnoService.ts]: Missing typing

File: api/emAnno/emAnnoService.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * @param annotationsPath
+ * @param body
+ * @returns {Promise<null>}
+ */
+export async function handlePut(annotationsPath, body) {
+
+    try {
+        logger.info('putting annotations', annotationsPath)
+        const response = await http.put(annotationsPath, body)
```
Comment (ronaldmansveld) [inline api/emAnno/emAnnoService.ts]: Missing typing

File: api/emAnno/emAnnoService.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+ * Delete Annotations
+ *
+ * @param annotationsPath
+ * @returns {Promise<null>}
+ */
+export async function handleDelete(annotationsPath) {
+
+    try {
+        logger.info('deleting annotations', annotationsPath)
+        const response = await http.delete(annotationsPath)
```
Comment (ronaldmansveld) [inline api/emAnno/emAnnoService.ts]: Missing typing

File: api/emAnno/emAnnoService.ts [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+}
+
+/**
+ * Delete Annotations
+ *
+ * @param annotationsPath
+ * @returns {Promise<null>}
+ */
+export async function handleDelete(annotationsPath) {
```
Comment (ronaldmansveld) [inline api/emAnno/emAnnoService.ts]: Missing typing

File: api/emAnno/emAnnoService.ts [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+/**
+ * Put Annotations
+ *
+ * @param annotationsPath
+ * @param body
+ * @returns {Promise<null>}
+ */
+export async function handlePut(annotationsPath, body) {
```
Comment (ronaldmansveld) [inline api/emAnno/emAnnoService.ts]: Missing typing

File: api/emAnno/emAnnoService.ts [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+/**
+ * Post Annotations
+ *
+ * @param annotationsPath
+ * @param body
+ * @returns {Promise<null>}
+ */
+export async function handlePost(annotationsPath, body) {
```
Comment (ronaldmansveld) [inline api/emAnno/emAnnoService.ts]: Missing typing

File: api/emAnno/index.spec.ts [hunk 12]
```diff
@@ -0,0 +1,167 @@
+/* tslint:disable:no-unused-expression no-var-requires */
```
Comment (ronaldmansveld) [inline api/emAnno/index.spec.ts]: As above on disabling no-unused-expression

File: package.json [hunk 13]
```diff
@@ -54,9 +54,10 @@
     "@angular/platform-browser": "^7.2.13",
     "@angular/platform-browser-dynamic": "^7.2.13",
     "@angular/router": "^7.2.13",
-    "@hmcts/ccd-case-ui-toolkit": "^2.59.1",
+    "@hmcts/ccd-case-ui-toolkit": "^2.62.16-RDM-5582-prerelease",
```
Comment (ronaldmansveld) [inline package.json]: Do we want to rely on a prerelease version in production code, or can we use a stable version?

File: api/amendedJurisdictions/index.ts (hunk 1 above)
Comment (adnanakgun) [reply api/amendedJurisdictions/index.ts]: Forgot to re-change base to master after rebase. This file is not part of this PR.

File: api/amendedJurisdictions/index.ts (hunk 2 above)
Comment (adnanakgun) [reply api/amendedJurisdictions/index.ts]: Forgot to re-change base to master after rebase. This file is not part of this PR.

File: package.json (hunk 13 above)
Comment (adnanakgun) [reply package.json]: That is the ticket.

File: src/app/components/media-viewer-wrapper/media-viewer-wrapper.component.scss [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+
+  .textLayer > div {
+    color: transparent;
+    position: absolute;
+    white-space: pre;
+    cursor: text;
+    -webkit-transform-origin: 0% 0%;
+    -moz-transform-origin: 0% 0%;
+    -o-transform-origin: 0% 0%;
+    -ms-transform-origin: 0% 0%;
```
Comment (ronaldmansveld) [inline src/app/components/media-viewer-wrapper/media-viewer-wrapper.component.scss]: Shouldn't need to use vendor prefixes in scss with angular, as these will get added during build based on the config in the browserslist file

File: src/app/components/media-viewer-wrapper/media-viewer-wrapper.component.scss [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+  .textLayer .highlight {
+    margin: -1px;
+    padding: 1px;
+
+    background-color: rgb(180, 0, 170);
+    border-radius: 4px;
+  }
+
+  .textLayer .highlight.begin {
+    border-radius: 4px 0px 0px 4px;
```
Comment (ronaldmansveld) [inline src/app/components/media-viewer-wrapper/media-viewer-wrapper.component.scss]: 0px values should be unitless

File: src/app/components/media-viewer-wrapper/media-viewer-wrapper.component.scss [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+    position: absolute;
+    left: 0px;
+    top: 100%;
+    right: 0px;
+    bottom: 0px;
+    z-index: -1;
+    cursor: default;
+    -webkit-user-select: none;
+    -ms-user-select: none;
+    -moz-user-select: none;
```
Comment (ronaldmansveld) [inline src/app/components/media-viewer-wrapper/media-viewer-wrapper.component.scss]: As above, plus this one is missing the rule without the vendor prefix

File: src/app/components/media-viewer-wrapper/media-viewer-wrapper.component.scss [hunk 17]
```diff
@@ -0,0 +1,220 @@
...
+  }
+
+  .textLayer .endOfContent.active {
+    top: 0px;
```
Comment (ronaldmansveld) [inline src/app/components/media-viewer-wrapper/media-viewer-wrapper.component.scss]: as above

File: src/app/components/media-viewer-wrapper/media-viewer-wrapper.component.ts [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+        private windowService: WindowService
+    ) {
+    }
+
+    async ngOnInit() {
+
+
+        const localStorageMedia = this.windowService.getLocalStorage(MEDIA_VIEWER);
+        if (localStorageMedia) {
+            const media = JSON.parse(localStorageMedia);
```
Comment (ronaldmansveld) [inline src/app/components/media-viewer-wrapper/media-viewer-wrapper.component.ts]: missing typing

File: src/app/services/ccd-config/ccd-case.config.ts [hunk 19]
```diff
@@ -105,4 +105,8 @@ export class AppConfig extends AbstractAppConfig {
   public getPaginationPageSize(): number {
     return this.config.pagination_page_size;
   }
+
+  public getAnnotationApiUrl() {
```
Comment (ronaldmansveld) [inline src/app/services/ccd-config/ccd-case.config.ts]: missing typing

**Pull request merged**